### PR TITLE
Attempt to parse GNU extension 1002.

### PR DESCRIPTION
**Pull request by alex**
Refs keybase/keybase-issues#1885

### CODE REVIEW

File: openpgp/s2k/s2k.go [hunk 1]
```diff
@@ -168,11 +175,26 @@ func parseGNUExtensions(r io.Reader) (f func(out, in []byte), err error) {
...
+		return nil, nil
+	} else if gnuExtType == 2 {
+		var lenBuf [1]byte
+		_, err = io.ReadFull(r, buf[:1])
+		if err != nil {
+			return
+		}
+
+		var ivBuf [16]byte
+		ivLen := minInt(lenBuf[0], 16)
```
Comment (maxtaco) [inline openpgp/s2k/s2k.go:188]: I thought `gpg` was performing a `max` here?

Comment (alex) [inline openpgp/s2k/s2k.go:188]: No, I had that backwards: http://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=blob;f=g10/parse-packet.c;h=c925e9440e371a223f90ac71d7379b2bfc706c40;hb=refs/heads/STABLE-BRANCH-2-0#l1993

Comment (maxtaco) [inline openpgp/s2k/s2k.go:188]: Ah, you're right. It's a `min`.  I'll merge and then tweak the style a bit, thanks!

**Comment by alex** [comment]
Thank you!

**Pull request merged**